bundle/edu.gemini.pot/src/main/scala/edu/gemini/spModel/target/env/Asterism.scala

```diff
@@ -190,4 +194,71 @@ object Asterism {
   def createSingleAsterism: Single = {
     Single(new SPTarget())
   }
+
+  def fromTypeAndTemplateTarget(
+    asterismType: AsterismType,
+    target:       SPTarget
+  ): Asterism = {
+
+    def targetPlus(am: Int): SPTarget = {
+      val res = new SPTarget()
+      res.setTarget(
+        Target.ra.mod(_.offset(Angle.fromArcmin(am.toDouble)), target.getTarget)
+      )
+      res
+    }
+
+    def coordinatesPlus(am: Int): SPCoordinates =
+      target.getCoordinates(None).fold(new SPCoordinates()) { c =>
+        new SPCoordinates(c.offset(Angle.fromArcmin(am.toDouble), Angle.zero))
+      }
+
+    def ghostTarget(sp: SPTarget): GhostTarget =
+      GhostTarget(sp, GuideFiberState.Enabled)
+
+    asterismType match {
+      case AsterismType.Single                              =>
+        Single(target)
+
+      case AsterismType.GhostSingleTarget                   =>
+        GhostAsterism.SingleTarget(ghostTarget(target), None)
+
+      case AsterismType.GhostDualTarget                     =>
+        GhostAsterism.DualTarget(
+          ghostTarget(targetPlus(1)),
```

Comment:
I guess this means we generate two targets and the one from the proposal becomes the base at the center?